app/services/puppet_fact_parser.rb

```diff
@@ -145,6 +145,22 @@ def boot_timestamp
     uptime_seconds.nil? ? nil : (Time.zone.now.to_i - uptime_seconds.to_i)
   end
 
+  def virtual
+    facts['is_virtual'] == "true" || facts['is_virtual'] || facts['virtual']
```

Comment:
`facts['virtual']` is a string (`kvm`, `physical`, ..) while the data type is a boolean. Since it can also be a string when it's not virtual (`physical`), I'd leave it out here.

Another thing I'd consider is dropping the support for string facts. If you do consider it supported, the `"false"` will be considered the boolean `true`. However, since Puppet 4 we can rely on proper data types. https://puppet.com/docs/puppet/4.10/lang_facts_and_builtin_vars.html#handling-boolean-facts-in-older-puppet-versions has a bit of background. We've dropped support for Puppet 3 so we don't need to care about it anymore.

This it can simply be:
```suggestion
  facts['is_virtual']
```